### ISSUE TITLE
yabai --load-sa fails to inject scripting-addition on macOS Monterey 12.3

### ISSUE THREAD

**Issue by kraeki** [koekeishiya/yabai#1231]
Hi gents

I am running the new MAC Book Pro (12.3).

### What I did
1. Install yabai 4.0.0
2. Disable SIP
3. Set boot-args

Still, when I try to load the scripting-addition it fails
```
$ sudo yabai --load-sa
yabai: scripting-addition failed to inject payload into Dock.app!

```

I also checked the architecture of the yabai binary etc. But looks good in yabai 4.0.0. 

I also tried to find Dock logs but there is nothing coming up in the Console.app. 
Do you have some pointers? 

### Further Logs
```
$ nvram boot-args
boot-args	-arm64e_preview_abi
```

```
$ file $(which yabai)
/opt/homebrew/bin/yabai: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64
- Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64
- Mach-O 64-bit executable arm64]
/opt/homebrew/bin/yabai (for architecture x86_64):	Mach-O 64-bit executable x86_64
/opt/homebrew/bin/yabai (for architecture arm64):	Mach-O 64-bit executable arm64
```

```
$ cat /Library/ScriptingAdditions/yabai.osax/Contents/Info.plist | grep -A 1 CFBundleVersion
<key>CFBundleVersion</key>
<string>2.0.4</string>
```

```
$ file /Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit dynamically linked shared library x86_64
- Mach-O 64-bit dynamically linked shared library x86_64] [arm64e:Mach-O 64-bit dynamically linked shared library arm64e
- Mach-O 64-bit dynamically linked shared library arm64e]
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload (for architecture x86_64):	Mach-O 64-bit dynamically linked shared library x86_64
/Library/ScriptingAdditions/yabai.osax/Contents/Resources/payload.bundle/Contents/MacOS/payload (for architecture arm64e):	Mach-O 64-bit dynamically linked shared library arm64e
```

```
$ file /Library/ScriptingAdditions/yabai.osax/Contents/MacOS/mach_loader
/Library/ScriptingAdditions/yabai.osax/Contents/MacOS/mach_loader: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64
- Mach-O 64-bit executable x86_64] [arm64e:Mach-O 64-bit executable arm64e
- Mach-O 64-bit executable arm64e]
/Library/ScriptingAdditions/yabai.osax/Contents/MacOS/mach_loader (for architecture x86_64):	Mach-O 64-bit executable x86_64
/Library/ScriptingAdditions/yabai.osax/Contents/MacOS/mach_loader (for architecture arm64e):	Mach-O 64-bit executable arm64e
```

**Comment by PaddiM8** [comment]
I'm having this issues as well on 12.3.1. SIP is fully disabled. I actually had this issue when I was on 11.3 as well though. I have used xorpse/yabai previously, and decided to change to koekeishiya/yabai 4.0.0 today before upgrading the OS. The problem persisted after the OS upgrade.

**Comment by VickyMerzOwn** [comment]
Well, I think I'm having a similar issue. When I run `yabai --check-sa` after both `yabai --install-sa` and `yabai --uninstall-sa` once, I get the same 0 exit code (no message is thrown). 

Also, even after I run `yabai --load-sa` and run `yabai -c ~/.yabairc` (to set the config file to "~/.yabairc") or for that matter only `yabai` (just to check if it works fine), I get the following message:
`yabai: could not acquire lock-file! abort..`

Worth noting, when I set other configurations with commands like `yabai -m config focus_follows_mouse autofocus`, it works as I would like it to.
Can't understand what I'm doing wrong here. For the record, I just started with yabai and installed it just today on MacOS Monterey 12.2.1 .

**Comment by madkinsz** [comment]
I'm also encountering this issue with the SIP partially disabled per the current installation guide. Unlike some of the other issues that report SA loading issues, this one does not include an additional message about the cause of failure.

@VickyMerzOwn 

> `yabai: could not acquire lock-file! abort..`

Generally indicates that you are running another instance of yabai, perhaps via the brew service? You can run `killall yabai` and `brew services stop yabai` and that should be resolved.

**Comment by VickyMerzOwn** [comment]
@madkinsz I did that. Understood the problem now, I guess.
But I'm facing this now.
Once I `brew services stop yabai`, it kills all yabai processes. Then, I ran `yabai --config ~/.yabairc`. This runs yabai after running the yabai config script. I realise this works as specified there. 
However, I expected it to set the specified file as the config file for whenever the yabai service starts.
Is there any way to make this happen?
